src/main/scala/nl/knaw/dans/api/sword2/DepositHandler.scala

```diff
@@ -228,11 +240,11 @@ object DepositHandler {
     } yield ()
   }
 
-  def checkBagVirtualValidity(bagDir: File)(implicit id: String, bagStoreBaseDir: File, bagStoreBaseUri: URI): Try[Unit] = {
+  def checkBagVirtualValidity(bagDir: File)(implicit id: String, bs: BagStoreBase): Try[Unit] = {
     log.debug(s"[$id] Verifying bag validity")
 
     val fetchItems = getFetchTxt(bagDir).map(_.asScala).getOrElse(Seq())
-    val fetchItemsInBagStore = fetchItems.filter(_.getUrl.startsWith(bagStoreBaseUri.toString))
+    val fetchItemsInBagStore = fetchItems.filter(bs.isBagStoreAware && _.getUrl.startsWith(bs.baseUrl))
```

Comment:
remove these lines here and replace line 277 (`val itemsToResolve` etc.) with the following:

```scala
val fetchItems = getFetchTxt(bagDir).map(_.asScala).getOrElse(Seq())
val (fetchItemsInBagStore, itemsToResolve) = fetchItems.partition(_.getUrl.startsWith(bs.baseUrl))

if (fetchItemsInBagStore.nonEmpty && !bs.isBagStoreAware)
  Failure(new IllegalArgumentException("blabla"))
else
```

also indent the for-comprehension to be in the else-branch